elements/camera-capture/camera-capture.js

```diff
@@ -0,0 +1,320 @@
+import { html, css, LitElement } from '../../node_modules/lit-element';
+
+import '../../node_modules/@material/mwc-icon';
+import '../../node_modules/@material/mwc-ripple';
+
+import './settings-pane.js';
+
+class CameraCapture extends LitElement {
+  static styles = css`
+    :host {
+      display: block;
+      width: 100vw;
+      height: 100vh;
+      overflow: hidden;
+    }
+
+    .hidden {
+      display: none !important;
+    }
+
+    #mainContent {
+      margin: 0 auto;
+    }
+
+    video {
+      display: inline;
+    }
+
+    video:hover {
+      cursor: pointer;
+    }
+
+    #resetButton {
+      position: relative;
+      padding: 15px;
+      color: white;
+      text-shadow: -1px 0 black, 0 1px black, 1px 0 black, 0 -1px black;
+      font-size: 18px;
+      background: none;
+      border: none;
+      z-index: 15;
+      outline: none;
+    }
+
+    .canvas-wrapper {
+      position: relative;
+      width: 100%;
+    }
+
+    .settings-wrapper {
+      position: absolute;
+      top: 0;
+      width: 100%;
+      height: 100%;
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      align-items: flex-end;
+    }
+
+    #cameraBar {
+      height: 100px;
+      background-color: transparent;
+      display: flex;
+      flex-direction: row;
+      flex-wrap: nowrap;
+      justify-content: space-around;
+      align-items: center;
+    }
+
+    #gallery {
+      width: 48px;
+      height: 48px;
+    }
+
+    #takePhotoButton {
+      height: 72px;
+      width: 72px;
+    }
+
+    #takePhotoButton mwc-icon {
+      --mdc-icon-size: 48px;
+    }
+
+    .camera-bar-icon {
+      height: 52px;
+      width: 52px;
+      display: flex;
+      justify-content: center;
+      outline: none;
+      border: 2px solid white;
+      border-radius: 52px;
+      color: white;
+      background-color: transparent;
+    }
+  `;
+
+  facingMode = "user";
+
+  _onResetClicked(e) {
```

Comment:
How about  reset-ing state per settings group ? Like it was before, so that we have more fine grained control of things we are resetting

Reply:
Done